app/com/linkedin/drelephant/exceptions/core/ExceptionFingerprintingRunner.java

```diff
@@ -99,20 +116,103 @@ private void saveDriverExceptionLogForExceptionFingerPrinting(List<ExceptionInfo
+        logger.error(" Error while storing 1/2 of the spark job exception ", pe1);
+        List<ExceptionInfo> firstException = createMixofException(exceptionInfoList, 2);
+        sparkJobException.exceptionLog = parseExceptions(firstException, false);
+        sparkJobException.save();
+        debugLog(" Final String to be saved in dB " + sparkJobException.exceptionLog);
+      }
+    }
+  }
+
+  /**
+   * Since we are not able to store all the exceptions in database , this method
+   * will create the mix of exceptions (including Executor and Driver)
+   * @param exceptionInfoList : List of exceptions .
+   * @param size : Maximum size of the list
+   * @return : Mix of exceptions
+   */
+  private List<ExceptionInfo> createMixofException(List<ExceptionInfo> exceptionInfoList, int size) {
+    List<ExceptionInfo> mixOfException = new ArrayList<>();
+    List<ExceptionInfo> executorExceptions = new ArrayList<>();
+    for (ExceptionInfo exceptionInfo : exceptionInfoList) {
+      if (exceptionInfo.getExceptionSource().name().equals(ExceptionInfo.ExceptionSource.EXECUTOR.name())) {
+        executorExceptions.add(exceptionInfo);
+      }
+    }
+    logger.info(" Size of executor exceptions "+executorExceptions.size());
+    for (int index = 0; index <= size / 2; index++) {
+      mixOfException.add(exceptionInfoList.get(index));
+    }
+    logger.info(" Size of mix exceptions "+mixOfException.size());
+    int max = Math.min(executorExceptions.size(), size - (size / 2));
+    for (int index = 0; index < max; index++) {
+      mixOfException.add(executorExceptions.get(index));
+    }
+    logger.info(" Size of mix exceptions "+mixOfException.size());
+    return mixOfException;
+  }
+
+  /**
+   *  User provided configuration is being used to set the number of
+   *  exceptions in DB
+   * @param exceptionInfoList : Exception List
+   * @param isFirstTime : Is the list parsed first time
+   * @return JSON Array as String containing exceptions
+   */
+  private String parseExceptions(List<ExceptionInfo> exceptionInfoList, boolean isFirstTime) {
+    ObjectMapper Obj = new ObjectMapper();
+    if (isFirstTime) {
+      Collections.sort(exceptionInfoList);
+    }
+    String exceptionInJson = null;
+    try {
+      exceptionInJson = Obj.writeValueAsString(exceptionInfoList.subList(0,
+          Math.min(exceptionInfoList.size(), NUMBER_OF_EXCEPTION_TO_PUT_IN_DB.getValue())));
+      logger.info(" Final Size of the exception List , which will be stored in db "+exceptionInJson);
+    } catch (IOException e) {
+      logger.error(" Exception while searlizing stack tract list to JSON ", e);
+    }
+    return exceptionInJson;
+  }
+
+  private String processLogSourceInformation(Map<String, String> logSourceInformation) {
+    StringBuilder logSource = new StringBuilder();
+    for (Map.Entry<String, String> entry : logSourceInformation.entrySet()) {
+      logSource.append("SOURCE " + entry.getKey() + "\tURL " + entry.getValue()).append("\n");
```

Comment:
This should be 2 different fields in DB, instead of 1, namely SOURCE and URL.

Reply:
But then we will be having multiple rows , one each source . If in future we will be having more source then we can move to multiple fields in the database . However , we have to see pros and cons for the same.

Reply:
It's better to keep data in structured format. Keeping in String and then serialization and deserialization add overhead. For a single failure, how many different sources can have ?

Reply:
Currently we have two sources , but latter it may be more  . For e.g Azkaban . Its still in structured format with tab seperation . Only thing is to have multiple rows for each source or to have single one .  As told , we can have multiple rows as suggested , but that will include redundancy in the data , thats the only thing .